ls1intum/Artemis
# [Bugfix] Fix issue with zero point exercises with bonus points

### Checklist
- [x] Client & Server: I added tests

### Motivation and Context


Resolves: #2538 

### Description

- Changed to correct getter
- Add special case when no maxPoints but bonusPoints are set
### Steps for Testing


1. Log in to Artemis
2. Navigate to Course Administration
3. Create a exercise like it is described in #2538. Programming exercise with maxPoints of 0 and bonusPoints of 100
5. Participate and correctly solve some tests (e.g. BubbleSort or MergeSort -> just add `Collections.sort(input)`)
6. Check that result string and result details are correct, for each successful test case you get 7,7 points -> score increases by 8%
6.1 1 test case -> 8% score, feedback element with 7,7 points in result popup
7. Play around with the grading, when giving bonus points for a test case, the points and score should increase accordingly. (see also screenshots below)

## **`Updated`**

**Important:** test the following 3 cases in 3 different exercises:

- [ ] 10 points + 0 bonus points
- [ ] 10 points + 10 bonus points
- [ ] 0 points + 10 bonus points

Test the following scenario: if a student e.g. has collected 30 points in a course, but the sum of normal points is 20, then the score on the start page of Artemis should show 150%. The score calculation is really critical and has to be absolutely correct!
Also make sure that the instructor page `course scores` includes the correct calculation based on the sum of normal points and the exported csv file is correct.

### Test Coverage



Nothing changed

### Screenshots


1 successful test case:
<img width="1421" alt="image" src="https://user-images.githubusercontent.com/41108487/102551025-22b2b380-40bf-11eb-916b-63dd0d261e25.png">

<img width="1424" alt="image" src="https://user-images.githubusercontent.com/41108487/102551057-2b0aee80-40bf-11eb-857d-7ee6a6d7bbc8.png">

2 successful test cases:
<img width="1410" alt="image" src="https://user-images.githubusercontent.com/41108487/102551157-52fa5200-40bf-11eb-8a7a-f734511b2df9.png">

<img width="1435" alt="image" src="https://user-images.githubusercontent.com/41108487/102551184-5c83ba00-40bf-11eb-9bf8-81652021d640.png">

2 successful test cases with grading adapt to give 50 bonus points for successful testBubbleSort and 70 bonus points for successful testMergeSort:
<img width="1413" alt="image" src="https://user-images.githubusercontent.com/41108487/102551366-a8cefa00-40bf-11eb-9474-428a4b3ae71c.png">

<img width="1408" alt="image" src="https://user-images.githubusercontent.com/41108487/102551406-b1bfcb80-40bf-11eb-9053-dde69ad171b1.png">
